pkg/openshift/rosa/cluster.go

```diff
@@ -0,0 +1,441 @@
+			return options, fmt.Errorf("subnet ids is required for hosted control plane clusters")
+		}
+	}
+
+	if options.accountRoles.controlPlaneRoleARN == "" {
+		return options, fmt.Errorf("iam role arn for control plane is required")
+	}
+
+	if options.accountRoles.installerRoleARN == "" {
+		return options, fmt.Errorf("iam role arn for installer is required")
+	}
+
+	if options.accountRoles.supportRoleARN == "" {
+		return options, fmt.Errorf("iam role arn for support role is required")
+	}
+
+	if options.accountRoles.workerRoleARN == "" {
+		return options, fmt.Errorf("iam role for worker role is required")
+	}
+
+	return options, nil
+}
+
+// createCluster handles sending the request to create the cluster
+func (r *Provider) createCluster(ctx context.Context, options *CreateClusterOptions) (string, error) {
+	options, err := validateCreateClusterOptions(options)
+	if err != nil {
+		return "", fmt.Errorf("cluster options validation failed: %v", err)
+	}
+
+	commandArgs := []string{"create", "cluster", "--output", "json", "--mode", "auto", "--yes"}
+	commandArgs = append(commandArgs, "--cluster-name", options.ClusterName)
+	commandArgs = append(commandArgs, "--channel-group", options.ChannelGroup)
+	commandArgs = append(commandArgs, "--compute-machine-type", options.ComputeMachineType)
+	commandArgs = append(commandArgs, "--machine-cidr", options.MachineCidr)
+	commandArgs = append(commandArgs, "--region", r.awsCredentials.Region)
+	commandArgs = append(commandArgs, "--version", options.Version)
+	commandArgs = append(commandArgs, "--replicas", fmt.Sprint(options.Replicas))
+	commandArgs = append(commandArgs, "--controlplane-iam-role", options.accountRoles.controlPlaneRoleARN)
+	commandArgs = append(commandArgs, "--role-arn", options.accountRoles.installerRoleARN)
+	commandArgs = append(commandArgs, "--support-role-arn", options.accountRoles.supportRoleARN)
+	commandArgs = append(commandArgs, "--worker-iam-role", options.accountRoles.workerRoleARN)
+
+	if options.Properties != "" {
+		commandArgs = append(commandArgs, "--properties", options.Properties)
+	}
+
+	if options.HostedCP {
+		commandArgs = append(commandArgs, "--hosted-cp")
+		commandArgs = append(commandArgs, "--oidc-config-id", options.oidcConfigID)
+		commandArgs = append(commandArgs, "--subnet-ids", options.subnetIDs)
+	}
+
+	if options.STS {
+		commandArgs = append(commandArgs, "--sts")
+	}
+
+	log.Printf("Creating cluster %q\n", options.ClusterName)
+
+	_, _, err = r.RunCommand(ctx, exec.CommandContext(ctx, r.rosaBinary, commandArgs...))
+	if err != nil {
+		return "", err
+	}
+
+	cluster, err := r.getCluster(ctx, options.ClusterName)
+	if err != nil {
+		return "", err
+	}
+
+	clusterID := cluster.ID()
+
+	log.Printf("Cluster %q created: %s", options.ClusterName, clusterID)
+
+	return clusterID, err
+}
+
+// getCluster gets the cluster the body
+func (r *Provider) getCluster(ctx context.Context, clusterName string) (*clustersmgmtv1.Cluster, error) {
```

Comment:
```suggestion
// findCluster gets the cluster the body
func (r *Provider) findCluster(ctx context.Context, clusterName string) (*clustersmgmtv1.Cluster, error) {
```

i think I'd reserve `get` for passing in the clusterID and returning the cluster object

Reply:
Sounds good by me. 38aa9853b7e033adc0ddd0c576caf65225fb1668